app/helpers/pipeline_outputs_helper.rb

```diff
@@ -1,12 +1,17 @@
 module PipelineOutputsHelper
   Client = Aws::S3::Client.new
 
-  def get_taxid_fasta(pipeline_output, taxid)
-    uri_parts = pipeline_output.sample.sorted_taxid_annotated_fasta_s3_path.split("/", 4)
+  def get_taxid_fasta(pipeline_output, taxid, tax_level, hit_type)
+    uri = pipeline_output.sample.s3_paths_for_taxon_byteranges[tax_level][hit_type]
+    uri_parts = uri.split("/", 4)
     bucket = uri_parts[2]
     key = uri_parts[3]
-    taxon_location = pipeline_output.taxon_byteranges.find_by(taxid: taxid)
-    return 'No records' if taxon_location.nil?
+    taxon_location = pipeline_output.taxon_byteranges.find_by(taxid: taxid, hit_type: hit_type)
+    if taxon_location.nil? && hit_type == 'NT' && tax_level == TaxonCount::TAX_LEVEL_SPECIES
+      taxon_location = pipeline_output.taxon_byteranges.find_by(taxid: taxid, hit_type: nil, tax_level: nil)
```

Comment:
can you just run a migration instead of having the code the code here?

Reply:
Removed it. There are no taxon_byteranges on production yet, so no need to handle the old case.